src/app/data-entries/dropdown/dropdown-sample-4/dropdown-sample-4.component.ts

```diff
@@ -1,36 +1,24 @@
-import { Component, OnInit, ViewChild } from "@angular/core";
+import { Component, ViewChild } from "@angular/core";
 import {
     ConnectedPositioningStrategy,
     IgxDropDownComponent,
-    IgxInputDirective,
-    IgxInputGroupComponent,
-    ISelectionEventArgs
+    IgxInputGroupComponent
 } from "igniteui-angular";
 @Component({
     // tslint:disable-next-line:component-selector
     selector: "app-dropdown-sample-4",
     styleUrls: ["./dropdown-sample-4.component.scss"],
     templateUrl: "./dropdown-sample-4.component.html"
 })
-export class DropDownSample4Component implements OnInit {
+export class DropDownSample4Component {
     @ViewChild(IgxDropDownComponent) public igxDropDown: IgxDropDownComponent;
     @ViewChild("inputGroup", { read: IgxInputGroupComponent}) public inputGroup: IgxInputGroupComponent;
-    @ViewChild("input", { read: IgxInputDirective })
-    public input: IgxInputDirective;
 
-    public items: any[] = [];
-    public value: string;
-
-    public ngOnInit() {
-        for (let i = 1; i < 4; i ++) {
-            const item = { field: "Option " + i };
-            this.items.push(item);
-        }
-    }
-
-    public onSelection(eventArgs: ISelectionEventArgs) {
-        this.value = eventArgs.newSelection.value;
-    }
+    public items: Array<{ field: string }> = [
+        { field: "Option 1" },
+        { field: "Option 2" },
+        { field: "Option 3" }
+    ];
 
     public openDropDown() {
```

Comment:
this will not work now. You cannot change the target position strategy when you are using IgxToggleAction. Also there is no need to check if drop down is collapsed - just call Open - IgxToggleDirective will check everything and will open the drop down only when necessary.

Reply:
This is a call to the drop down's open method, which positions the content directly through the overlay service, bypassing the `igxToggleAction` directive. It will display the drop down properly, though the method itself is not really elegant - as we can't access the `igxToggleAction`'s overlay settings